backend/src/test/java/de/neuefische/backend/controller/ItemControllerTest.java

```diff
@@ -0,0 +1,172 @@
+    void createNewItem_whenLoggedInAsUser_shouldReturnCreatedItem() throws Exception {
+        Item testItem = new Item("1", "test", 1.0, "test", null, "test");
+        final String json = new ObjectMapper().writerWithDefaultPrettyPrinter()
+                .writeValueAsString(testItem);
+
+        mockMvc.perform(post("/api/items")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(json))
+                .andExpect(status().isOk())
+                .andExpect(content().json(json));
+
+        assertThat(itemRepository.findById("1")).isPresent();
+        assertThat(itemRepository.findById("1").get()).isEqualTo(testItem);
+    }
+
+    @Test
+    void getAll_whenNotLoggedIn_Return401() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/items"))
+                .andExpect(status().isUnauthorized()
+                );
+    }
+
+    @Test
+    @WithMockUser(username = "testuser")
+    void getAll_whenLoggedInAsUser_shouldReturnItems() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/items"))
+                .andExpectAll(
+                        status().isOk(),
+                        content().json(
+                                "[]",
+                                true
+                        )
+                );
+    }
+
+    @Test
+    void getItemById_whenNotLoggedIn_shouldReturn401() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/items/1"))
+                .andExpect(status().isUnauthorized()
+                );
+    }
+
+    @Test
+    @WithMockUser(username = "testuser")
+    void getItemById_whenLoggedInAsUser_shouldReturnItem() throws Exception {
+        String itemAsJson = """
+                {"id":"1","name":"test","price":1.0,"description":"test","image":null,"category":"test"}""";
+
+        mockMvc.perform(post("/api/items")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(itemAsJson))
+                .andExpect(status().isOk())
+                .andExpect(content().string(itemAsJson));
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/items/1"))
+                .andExpectAll(
+                        status().isOk(),
+                        content().json(
+                                itemAsJson,
+                                true
+                        )
+                );
+    }
+
+    @Test
+    void updateItem_whenNotLoggedIn_shouldReturn401() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.put("/api/items/1"))
+                .andExpect(status().isUnauthorized()
+                );
+    }
+
+    @Test
+    @WithMockUser(username = "testuser")
+    void updateItem_whenLoggedInAsUser_shouldReturnItem() throws Exception {
+        String itemAsJson = """
+                {"id":"1","name":"test","price":1.0,"description":"test","image":null,"category":"test"}""";
+
```

Comment:
hier würde ich auch in meheren zeilen teilen

Reply:
![image](https://user-images.githubusercontent.com/118479718/215027940-07c6e7d2-b40a-4a1d-b7a8-5a78a977f118.png)
Wenn ich das nicht in eine Zeile schreibe, bekomme ich hier einen Fehler geworfen. Anders formatiert finde ich es auch besser, aber das funktioniert im Test nicht. :/